### ISSUE TITLE
ROADMAP

### ISSUE THREAD

**Issue by asturur** [fabricjs/fabric.js#8316]
## Roadmap 🚀

This Pinned list will contain an updated list of things that we want or we are working on.
Completed items will move in issue comments below. There isn't an ETA for any feature, and 1 or more developer may be working on them.

Those two feature are currently worked on and have priority above anything else:
- Major release for TS and modern JS migration with change of build system and necessary breaking changes
- Finishing and polishing the group rewrite

The following list of feature are long standing items ( 2+ years each ) and in a perfect world we want to complete them before any other large refactor that could make them harder, or since introucing them later could make a refactor less effective.
Those listed feature are complex ( apart the filter one, that is easy) and touch many part of fabric codebase.

- Enable canvas viewport rotation https://github.com/fabricjs/fabric.js/pull/4624
- Enable MASK feature with SVG import https://github.com/fabricjs/fabric.js/pull/6310
- Enable pixel fitlers to any object https://github.com/fabricjs/fabric.js/pull/6311
- Improve/solve Sharpness of clipPaths
- Enable rendering of partial area of the screen
- Move controls out of the main canvas ( allow for change of selection without full re-rendering )
- Change the brush api to be more flexible and allow different type of brushes

If you are a contributor and want to propose a new feature, do it, there is always space for:

- new pixel filters
- new controls actions
- new brushes

**Comment by asturur** [comment]
Reserved for completed items

**Comment by usamshen** [comment]
@asturur great work! 

do you and @ShaMan123 have some regular meetings to discuss the fabricjs product vision and roadmap, if yes, i would like to join the discussion if possible.

**Comment by annibuliful** [comment]
> @asturur great work!
> 
> do you and @ShaMan123 have some regular meetings to discuss the fabricjs product vision and roadmap, if yes, i would like to join the discussion if possible.

Me too

**Comment by asturur** [comment]
Regarding meetings:
We still discuss how to bring this to the finish line mostly, and we are gearing for smooth collaboration.
I'm not opposed to open meetings if timezone allow it, if we want to discuss features and we can make point for the longevity of the library rathern than short term needs i m all for it.

**Comment by dubem-design** [comment]
Am it for it too, Maybe a Discord, Slack whatever the approach

**Comment by ShaMan123** [comment]
@asturur I want to add refactoring ObjectGeometry, making sense out of coords, deciding in which plane the exist and make it accessible. I still can't understand what is going on there. And I am sure there are bugs with group that aren't known since they remain untested.

**Comment by ShaMan123** [comment]
I will add that we can use planeChange to redo most of geometry with ease I believe

**Comment by asturur** [comment]
@ShaMan123 making oCoords/aCoords simpler has been a long standing issue, but i m afraid we can't do it now.
Is nearly one year we started this migration and we are not done yet. We do continue to stuff in incremental changes ( some are simplifications, some are improvements, some are just changes ) and we never finish.

**Comment by ShaMan123** [comment]
I am shocked a year is about to pass

**Comment by faizux** [comment]
when do you expect to release alpha v6?

**Comment by Smrtnyk** [comment]
> when do you expect to release alpha v6?

it is already in 6.0.0-beta19 and it is released, I have been using beta for a while now

**Comment by faizux** [comment]
> > when do you expect to release alpha v6?
> 
> it is already in 6.0.0-beta19 and it is released, I have been using beta for a while now

Yes, I'm aware of that. Currently, I'm using version 5.3.0 (RTL). However, I'm looking for stable release.

**Comment by asturur** [comment]
I m doing an effort to create a project template that works as roadmap container.
Bugfix do not need a roadmap, everything else will need a process.
We don't accept any new feature or refactor for v6 and we aim just at releasing it.
Every future change will be proposed with a description, we will try to collect sideeffects and opinions from developers and only after move to implementation or park the idea.
i started one as an example that is not going to happen this week or next week or this month because is another breaking change, an example of the description is here: https://github.com/fabricjs/fabric.js/discussions/9736 and as soon as i think i added all i can add i will open for comments.
I will do similar discussions for the items in the roadmap so that everyone has enough examples on what we look for when we think at changes.

**Comment by asturur** [comment]
I would also like to use this pinned issue to announce and link new discussions and progress, so people have ONE issue to monitor, so let's avoid spam if possible or questions

**Comment by JoyNop** [comment]
We want to use fabricjs to develop in new projects. Is V6 ready to release the stable release.? If we use the beta version  now, will it have any impact on the future?

**Comment by asturur** [comment]
Currently there is one large bug in v6 we want to fix and then should be released. All the rest will be bugfixes with no breaking changes. Other major versions could follow, maybe quickly maybe not, but we will try to have a single breaking change ( or feature)  per major version, no large reworks like this

**Comment by asturur** [comment]
Another small proposal https://github.com/fabricjs/fabric.js/discussions/9737

**Comment by zacol** [comment]
From what I've been able to find out, there were once plans to add support for layers. The mechanism we know from Konva.js. Here one of the references on the subject  https://github.com/fabricjs/fabric.js/issues/7692 Did this feature land in version 6?
